src/resource_pool_texture.hpp

```diff
@@ -32,9 +32,10 @@ namespace wr
 		TexturePool(TexturePool&&) = delete;
 		TexturePool& operator=(TexturePool&&) = delete;
 
-		[[nodiscard]] TextureHandle Load(std::string_view path, bool srgb, bool generate_mips);
-		[[nodiscard]] TextureHandle LoadFromMemory(char* data, int width, int height, TextureType type, bool srgb, bool generate_mips);
-		[[nodiscard]] TextureHandle LoadFromMemory(char* data, int width, int height, std::string texture_extension, bool srgb, bool generate_mips);
+		[[nodiscard]] virtual TextureHandle LoadFromFile(std::string_view path, bool srgb, bool generate_mips) = 0;
+		[[nodiscard]] virtual TextureHandle LoadFromCompressedMemory(char* data, size_t width, size_t height, std::string& texture_extension, bool srgb, bool generate_mips);
+		[[nodiscard]] virtual TextureHandle LoadFromCompressedMemory(char* data, size_t width, size_t height, TextureType type, bool srgb, bool generate_mips) = 0;
+		[[nodiscard]] virtual TextureHandle LoadFromRawMemory(char* data, size_t width, size_t height, bool srgb, bool generate_mips) = 0;
 		[[nodiscard]] virtual TextureHandle CreateCubemap(std::string_view name, uint32_t width, uint32_t height, uint32_t mip_levels, Format format, bool allow_render_dest) = 0;
 		[[nodiscard]] virtual TextureHandle CreateTexture(std::string_view name, uint32_t width, uint32_t height, uint32_t mip_levels, Format format, bool allow_render_dest) = 0;
 		virtual void Unload(uint64_t texture_id) = 0;
```

Comment:
Why does this use the id and not the entire handle :/

Reply:
Oversight! changed it

Reply:
Changed this, and much more